src/content/platform-integration/ios/app-extensions.md

```diff
@@ -117,207 +212,353 @@ Choose one of the following sources for your data.
+        self.extensionContext?.cancelRequest(
+            withError: NSError(domain: Bundle.main.bundleIdentifier!, code: 0))
+    }
+}
+```
+
+{% endtab %}
+
+{% tab "UIKit-ObjC" %}
+
+```objc title="ShareViewController.h"
+#import <UIKit/UIKit.h>
+#import <Flutter/Flutter.h>
+
+@interface ShareViewController : UIViewController
+
+@end
+```
+
+```objc title="ShareViewController.m"
+#import "ShareViewController.h"
+@import Flutter;
+
+@implementation ShareViewController
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self showFlutter];
+}
+
+- (void)showFlutter {
+    FlutterEngine *flutterEngine = [[FlutterEngine alloc] initWithName:@"my flutter engine"];
+    [flutterEngine run];
+    FlutterViewController *flutterViewController =
+            [[FlutterViewController alloc] initWithEngine:flutterEngine nibName:nil bundle:nil];
+    [self addChildViewController:flutterViewController];
+    [self.view addSubview:flutterViewController.view];
+    flutterViewController.view.frame = self.view.bounds;
+}
+
+- (void)viewDidDisappear:(BOOL)animated {
+    [super viewDidDisappear:animated];
+    [self.extensionContext cancelRequestWithError:[NSError errorWithDomain:NSBundle.mainBundle.bundleIdentifier code:0 userInfo:nil]];
+}
+@end
+```
+
+{% endtab %}
+
+{% endtabs %}
+
+8.  [Test your app with the simulator][].
+
+[Add an extension to your Flutter app]: #add-extension
+[Share]: https://developer.apple.com/library/archive/documentation/General/Conceptual/ExtensibilityPG/Share.html
+[Test your app with the simulator]: #test-extensions
+
+
+### Register plugins
+
+Use the following steps to register plugins for
+an app extension. In this example, the app extension
+scheme is called `ShareExtension`, the Flutter app scheme is
+called `Runner`, and the Flutter app is called
+`Example App With Extension`:
+
+1.  [Add an extension to your Flutter app][] if you haven't
+    already done so.
+
+1.  In Xcode, add `GeneratedPluginRegistrant.m` to the
+    app extension target.
+
+    *   Open the **project navigator**.
+
+    *   In the main window under **TARGETS**, select the
+        scheme called **ShareExtension**.
+
+    *   Open the **Build Phases** tab.
+
+    *   Expand **Compile Sources**.
+
+    *   Click **+**.
+
+    *   From the list in the
+        _Choose item to add_ dialog box, select
+        **GeneratedPluginRegistrant.m**.
+
+    *   Click **Add**.
+
+1.  (Swift only) In Xcode, update the
+    `SWIFT_OBJC_BRIDGING_HEADER` build setting.
+
+    *   Open the **project navigator**.
+
+    *   In the main window under **TARGETS**, select the
+        scheme called **ShareExtension**.
+
+    *   Open the **Build Settings** tab.
+
+    *   Navigate to **Swift Compiler - General** and make
+        sure that the following key and value exist:
+        
+        ```console
+        Objective-C Bridging Header: Runner/Runner-Bridging-Header.h
+        ```
+
+1.  In Xcode, update the code for `ShareViewController`
```

Comment:
@hellohuanlin look carefully over this step and the code for both Swift and Objective-C. I am not sure what I have here is correct.

Reply:
yep it's correct

Reply:
Awesome!